sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobServiceVersionPolicy.java

```diff
@@ -0,0 +1,54 @@
+package com.azure.storage.blob.implementation.util;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.storage.blob.BlobServiceVersion;
+import com.azure.storage.common.implementation.Constants;
+import reactor.core.publisher.Mono;
+
+import static com.azure.storage.common.implementation.StorageImplUtils.throwIfContainsHeader;
+
+public class BlobServiceVersionPolicy implements HttpPipelinePolicy {
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        HttpHeaders requestHeaders = context.getHttpRequest().getHeaders();
+        BlobServiceVersion serviceVersion = toServiceVersion(
+            requestHeaders.getValue(Constants.HeaderConstants.SERVICE_VERSION));
+        String ANY_BLOB_API = "any blob API";
+
+        if (serviceVersion.ordinal() == BlobServiceVersion.getLatest().ordinal()) {
+            return next.process();
+        }
+
+        // 2019_07_07 check
+        if (serviceVersion.ordinal() < BlobServiceVersion.V2019_07_07.ordinal()) {
+            // Encryption scope
+            String CREATE_CONTAINER = "create container";
+            throwIfContainsHeader(requestHeaders, "x-ms-default-encryption-scope", CREATE_CONTAINER,
+                serviceVersion.getVersion());
+            throwIfContainsHeader(requestHeaders, "x-ms-deny-encryption-scope-override", CREATE_CONTAINER,
+                serviceVersion.getVersion());
+            throwIfContainsHeader(requestHeaders, "x-ms-encryption-scope", ANY_BLOB_API, serviceVersion.getVersion());
+
+            // Managed disk range diff
+            throwIfContainsHeader(requestHeaders, "x-ms-previous-snapshot-url", ANY_BLOB_API,
+                serviceVersion.getVersion());
+        }
+
+        return next.process();
+    }
+
+    private static BlobServiceVersion toServiceVersion(String version) {
+        if (version.equals(BlobServiceVersion.V2019_02_02.getVersion())) {
```

Comment:
Should check that `BlobServiceVersion.getVersion` equals `version`.

```java
BlobServiceVersion.V2019_02-02.getVersions().equals(version)
```

This is better since the enum has to be set but `version` may not be set (though I think it always will be).